incubator/hnc/pkg/reconcilers/hierarchical_namespace_test.go

```diff
@@ -56,15 +57,79 @@ var _ = Describe("Hierarchy", func() {
 	It("should set the self-serve subnamespace's owner annotation to the current namespace", func() {
 		Eventually(getNamespaceAnnotation(ctx, barName, api.AnnotationOwner)).Should(Equal(fooName))
 	})
+
+	It("should set the hns.status.state to Ok if the above tests all pass", func() {
```

Comment:
What do you mean by "the above tests?" All the tests are supposed to be independent.

Re-reading this, I think all the tests from lines 35-63 should be combined into one big test, and maybe line 31-32 from `BeforeEach` as well.